dangerfile.js

```diff
@@ -86,11 +86,15 @@ const noFlowFiles = newJsFiles
 
 raiseIssueAboutPaths(warn, noFlowFiles, '@flow');
 
+// based on assumptions from https://github.com/nodejs/node/wiki/ES6-Module-Detection-in-Node
+// basically it needs to contain at least one line with import or export at the
+// beginning of it, followed by a space; any content thereafter doesn't matter
+const esModuleRegex = /^(import|export)\s/g;
```

Comment:
Flow uses [`import type`/`export type`](https://flowtype.org/docs/syntax.html#importing-and-exporting-types) syntax, so I think you can safely match it.